pkg/scaffold/v2/crd_editor_rbac.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"path/filepath"
+	"strings"
+	"fmt"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/input"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/v1/resource"
+	"github.com/gobuffalo/flect"
+)
+
+var _ input.File = &Group{}
+
+// CRD Editor role scaffolds the config/rbca/<kind>_editor_role.yaml
+type CRDEditorRole struct {
+	input.Input
+	// Plural is the plural lowercase of kind
+	Plural string
```

Comment:
It seems you don't need to do it yourself. Using https://github.com/kubernetes-sigs/kubebuilder/blob/eb3d7967cae29ba8bcf6893f359621551523bdeb/pkg/scaffold/v1/resource/resource.go#L46 should be fine. And we already populate it in https://github.com/kubernetes-sigs/kubebuilder/blob/eb3d7967cae29ba8bcf6893f359621551523bdeb/pkg/scaffold/v1/resource/resource.go#L67-L69

Reply:
Are you suggesting not declare the in the struct use to create the template? 

However, note that the plural and resource are required. See that it will be undefined if we remove it since the struct use has no Plural attribute. 

```
2019/11/11 22:45:48 error scaffolding APIs: template: *v2.CRDEditorRole:10:7: executing "*v2.CRDEditorRole" at <.Plural>: can't evaluate field Plural in type *v2.CRDEditorRole
make[1]: *** [generate-testdata] Error 1
make: *** [generate] Error 2

```

Also, see that the other scaffolds are also declaring them in the same way

https://github.com/kubernetes-sigs/kubebuilder/blob/100b673ad608e54cab6e5d2aaca6d98301b4ca22/pkg/scaffold/v2/crd/kustomization.go#L63-L64

https://github.com/kubernetes-sigs/kubebuilder/blob/cd0b10ff1a7966ce3d449ae9fbb8d7029259bc31/pkg/scaffold/v2/crd/enablewebhook_patch.go#L35

Reply:
{{ .Resource.Resource }} should be the plural of the resources. e.g. https://github.com/kubernetes-sigs/kubebuilder/blob/cd0b10ff1a7966ce3d449ae9fbb8d7029259bc31/pkg/scaffold/v2/crd/enablewebhook_patch.go#L59

Viewer role should look like:
```
type CRDViewerRole struct {
	input.Input

	// Resource is a resource in the API group
	Resource *resource.Resource
}

// GetInput implements input.File
func (g *CRDViewerRole) GetInput() (input.Input, error) {
	if g.Path == "" {
		g.Path = filepath.Join("config", "rbac", fmt.Sprintf("%s_viewer_role.yaml", strings.ToLower(g.Resource.Kind)))
	}
	g.TemplateBody = crdRoleViewerTemplate
	return g.Input, nil
}

// Validate validates the values
func (g *CRDViewerRole) Validate() error {
	return g.Resource.Validate()
}

var crdRoleViewerTemplate = `
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: {{ lower .Resource.Kind }}-viewer-role
rules:
- apiGroups:
  - {{ .Resource.Group }}.{{ .Domain }}
  resources:
  - {{ .Resource.Resource }}  # <==== this should the plural of your resources
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - {{ .Resource.Group }}.{{ .Domain }}
  resources:
  - {{ .Resource.Resource }}/status
  verbs:
  - get
`
```

Reply:
Catcher 👍 Thank you. doing it now.

Reply:
Done 👍